src/pip/_internal/operations/build/wheel_legacy.py

```diff
@@ -68,6 +69,18 @@ def build_wheel_legacy(
 
     Returns path to wheel if successfully built. Otherwise, returns None.
     """
+    deprecated(
+        reason=f"Building {name!r} using the legacy setup.py bdist_wheel mechanism.",
+        replacement=(
+            "to use the standardized build interface by "
+            "setting the `--use-pep517` option, "
+            "(possibly combined with `--no-build-isolation`), "
+            "or adding a `pyproject.toml` file to the source tree"
```

Comment:
Quick question, is it enough for a project to have a `pyproject.toml` for pip to use PEP-517 or does it have to have a (valid?)  `build-system` section"?

I ask because I have seen projects have `pyproject.toml` files with only `tool.[tool-name]` sections in them.

Reply:
> Quick question, is it enough for a project to have a `pyproject.toml` for pip to use PEP-517 or does it have to have a (valid?)  `build-system` section?

Yes, in that case it uses the default build backend with build isolation by default.